plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx

```diff
@@ -90,20 +55,26 @@ export const TechDocsReaderPage = ({
   const outlet = useOutlet();
   const entityName = entityRef ?? { kind, name, namespace };
 
+  // children is only going to be defined if no "element" prop is provided to the route
   if (!children) {
-    const childrenList = outlet ? Children.toArray(outlet.props.children) : [];
+    // the children of the outlet will always just be one component
+    // therefore we need to support either custom reader page OR addons (using our default reader page) and not both
+    const outletChildrenList = Children.toArray(outlet?.props.children);
 
-    const page = childrenList.find(child => {
-      const { type } = child as Extension;
-      return !type?.__backstage_data?.map?.get(TECHDOCS_ADDONS_WRAPPER_KEY);
+    const hasAddons = outletChildrenList.find(node => {
+      return getComponentData(node, TECHDOCS_ADDONS_WRAPPER_KEY);
     });
 
-    return (
-      (page as JSX.Element) || (
-        <TechDocsReaderPageProvider entityName={entityName}>
-          <TechDocsReaderLayout />
-        </TechDocsReaderPageProvider>
-      )
+    return hasAddons || !outletChildrenList.length ? (
```

Comment:
we want to render the "default" reader page when ever we have addons available (because the default reader has addon supprt) OR no custom reader is provided (because the default reader works without addons too. Does that make sense? Feel free to suggest how I can make it clearer. Is `hasAddons || outletChildrenList.length === 0` clearer?

Reply:
Got it, maybe not needed anymore if we can fix this case:
https://github.com/backstage/backstage/pull/10713#discussion_r846061727